WordPress/Classes/ViewRelated/Gutenberg/EditorMediaUtility.swift

```diff
@@ -147,31 +147,49 @@ class EditorMediaUtility {
         return imageDownload
     }
 
-    static func fetchRemoteVideoURL(for media: Media, in post: AbstractPost, completion: @escaping ( Result<(videoURL: URL, posterURL: URL?), Error> ) -> Void) {
-        guard let videoPressID = media.videopressGUID else {
-            //the site can be a self-hosted site if there's no videopressGUID
-            if let videoURLString = media.remoteURL,
-                let videoURL = URL(string: videoURLString) {
-                completion(Result.success((videoURL: videoURL, posterURL: nil)))
-            } else {
+    static func fetchRemoteVideoURL(for media: Media, in post: AbstractPost, withToken: Bool = false, completion: @escaping ( Result<(URL), Error> ) -> Void) {
+        // Return the attachment url it it's not a VideoPress video
+        if media.videopressGUID == nil {
+            guard let videoURLString = media.remoteURL, let videoURL = URL(string: videoURLString) else {
                 DDLogError("Unable to find remote video URL for video with upload ID = \(media.uploadID).")
                 completion(Result.failure(NSError()))
+                return
+            }
+            completion(Result.success(videoURL))
+        }
+        else {
+            fetchVideoPressMetadata(for: media, in: post) { result in
+                switch result {
+                case .success((let metadata)):
+                    guard let originalURL = metadata.originalURL else {
+                        DDLogWarn("Failed getting original URL for media with upload ID: \(media.uploadID)")
+                        return
```

Comment:
👋 @fluiddot will it be an issue if we don't call the completion closure in this case?

Reply:
> 👋 @fluiddot will it be an issue if we don't call the completion closure in this case?

Suggestion applied in https://github.com/wordpress-mobile/WordPress-iOS/pull/20237/commits/e8d43dcff862a469d8ffb34d3364a72cb4bea765.

> I understand that we shouldn't instantiate NSError empty, so I'm going to create a generic error for these cases.

Update applied in https://github.com/wordpress-mobile/WordPress-iOS/pull/20237/commits/cb453304a4d8d6c680a5a1b91a05edc807d08a1d.

Reply:
Thanks @fluiddot!